src/main/java/gregtech/common/metatileentities/multi/MetaTileEntityFissionReactor.java

```diff
@@ -0,0 +1,1104 @@
+                    }
+
+                    for (IFuelRodHandler fuelImport : this.getAbilities(MultiblockAbility.IMPORT_FUEL_ROD)) {
+                        if (fissionReactor.needsOutput) {
+                            ((MetaTileEntityFuelRodImportBus) fuelImport).getExportHatch(this.height - 1)
+                                    .getExportItems().insertItem(0,
+                                            OreDictUnifier.get(OrePrefix.fuelRodHotDepleted,
+                                                    fuelImport.getPartialFuel()),
+                                            false);
+                            this.fissionReactor.fuelMass -= 60;
+                        }
+                        if (canWork) {
+                            fuelImport.getStackHandler().extractItem(0, 1, false);
+                            needsReset |= fuelImport.setPartialFuel(fuelImport.getFuel());
+                            this.fissionReactor.fuelMass += 60;
+                        }
+                    }
+                    if (canWork) {
+                        fissionReactor.needsOutput = true;
+                        this.fissionReactor.fuelDepletion = 0.;
+
+                        if (needsReset) {
+                            fissionReactor.computeGeometry();
+                        }
+                    } else {
+                        this.unlockAll();
+                    }
+
+                }
+            }
+            this.updateReactorState(isFlowingCorrectly ? flowRate : 0);
+
+            this.syncReactorStats();
+
+            boolean melts = this.fissionReactor.checkForMeltdown();
+            boolean explodes = this.fissionReactor.checkForExplosion();
+            double hydrogen = this.fissionReactor.accumulatedHydrogen;
+            if (melts) {
+                this.performMeltdownEffects();
+            }
+            if (explodes) {
+                this.performPrimaryExplosion();
+                if (hydrogen > 1) {
+                    this.performSecondaryExplosion(hydrogen);
+                }
+            }
+        }
+    }
+
+    protected void performMeltdownEffects() {
+        this.unlockAll();
+        BlockPos.MutableBlockPos pos = new BlockPos.MutableBlockPos(this.getPos());
+        pos = pos.move(this.getFrontFacing().getOpposite(), diameter / 2);
+        for (int i = 0; i <= this.heightBottom; i++) {
+            this.getWorld().setBlockState(pos.add(0, -i, 0), Materials.Corium.getFluid().getBlock().getDefaultState());
+            this.getWorld().setBlockState(pos.add(1, -i, 0), Materials.Corium.getFluid().getBlock().getDefaultState());
+            this.getWorld().setBlockState(pos.add(-1, -i, 0), Materials.Corium.getFluid().getBlock().getDefaultState());
+            this.getWorld().setBlockState(pos.add(0, -i, 1), Materials.Corium.getFluid().getBlock().getDefaultState());
+            this.getWorld().setBlockState(pos.add(0, -i, -1), Materials.Corium.getFluid().getBlock().getDefaultState());
```

Comment:
Use `MutableBlockPos#move` instead of add. Move it to the location, set the block, then move back. `add` creates a new `BlockPos`.